packages/library-base/src/commonMain/kotlin/io/realm/Configuration.kt

```diff
@@ -44,6 +44,17 @@ public fun interface CompactOnLaunchCallback {
     public fun shouldCompact(totalBytes: Long, usedBytes: Long): Boolean
 }
 
+/**
+ * This interface is used to write data to a Realm file when the file is first created.
+ * It will be used in a way similar to using [Realm.writeBlocking].
+ *
+ * Note that writing data to a Realm file will involve IO, so it should generally only be done as
+ * part of opening a Realm on a background thread.
+ */
+public fun interface InitialDataCallback {
+    public fun MutableRealm.write()
```

Comment:
I think we need some explicit documentation of the actual method in the interface too. I like how this looks when using it, but it is a bit tricky to see how it is used it you are not very familiarized with our API. Maybe even more explicit if it included a small example to make it obvious how to use the `MutableRealm`-receiver.